src/briefcase/bootstraps/base.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Any, TypedDict
+
+
+class AppContext(TypedDict):
+    formal_name: str
+    app_name: str
+    class_name: str
+    module_name: str
+    project_name: str
+    description: str
+    author: str
+    author_email: str
+    bundle: str
+    url: str
+    license: str
+    briefcase_version: str
+    template_source: str
+    template_branch: str
+
+
+class BaseGuiBootstrap(ABC):
+    """Definition for a plugin that defines a new Briefcase app."""
+
+    fields: list[str] = [
+        "app_source",
+        "app_start_source",
+        "pyproject_table_briefcase_additional",
+        "pyproject_table_briefcase_app_additional",
+        "pyproject_requires",
+        "pyproject_test_requires",
+        "pyproject_table_macOS",
+        "pyproject_table_linux",
+        "pyproject_table_linux_system_debian",
+        "pyproject_table_linux_system_rhel",
+        "pyproject_table_linux_system_suse",
+        "pyproject_table_linux_system_arch",
+        "pyproject_table_linux_appimage",
+        "pyproject_table_linux_flatpak",
+        "pyproject_table_windows",
+        "pyproject_table_iOS",
+        "pyproject_table_android",
+        "pyproject_table_web",
+        "pyproject_extra_content",
+    ]
```

Comment:
Interested in feedback of prefacing many of these with the word `pyproject`.

My goal was to be as explicit as possible about what these fields are used for. A potential concern is this will create complications if a "new project" template doesn't use `pyproject.toml` in the future....since "pyproject" wouldn't really be an appropriate preface then. However, I think using a new preface (and therefore, all entirely new context fields) makes sense then to clearly communicate these fields are being used completely differently now (even if they are conceptually doing the same thing for the project).